src/TMap.cpp

```diff
@@ -3080,10 +3080,11 @@ std::pair<bool, QString> TMap::readJsonMapFile(const QString& source, const bool
         TArea* pArea = new TArea(this, pNewRoomDB);
         auto [id, name] = pArea->readJsonArea(mapObj.value(QLatin1String("areas")).toArray(), i);
         ++mProgressDialogAreasCount;
-        if (incrementJsonProgressDialog(false, true, 0)) {
-            if (allowUserCancellation) {
-                abort = true;
-            }
+        if (incrementJsonProgressDialog(false, true, 0) && allowUserCancellation) {
+            abort = true;
+            // Since we are break-ing and not going to store the TArea we just
+            // instantiated with new we need to clean it up:
```

Comment:
Let's save the comments for the 'why', not the 'what'? It's clear why we're deleting the area here.

Reply:
> Let's save the comments for the 'why'**A**, not the 'what'**B**? It's clear *why***C** we're deleting the area here.

:confused: I do not follow what you are asking for/saying? It wasn't clear to *me* that a `delete` was needed until Coverity pointed out the circumstances when it would be...